core/src/main/scala/org/http4s/parser/AcceptEncodingHeader.scala

```diff
@@ -11,22 +11,21 @@
 package org.http4s
 package parser
 
-import org.http4s.ContentCoding.ContentCodingParser
+import cats.parse.Parser1
+import cats.syntax.either._
 import org.http4s.headers.`Accept-Encoding`
-import org.http4s.internal.parboiled2._
 
 private[parser] trait AcceptEncodingHeader {
   def ACCEPT_ENCODING(value: String): ParseResult[`Accept-Encoding`] =
-    new AcceptEncodingParser(value).parse
+    acceptEncodingParser.parseAll(value).leftMap { e =>
+      ParseFailure("Invalid Accept Encoding header", e.toString)
+    }
 
-  private class AcceptEncodingParser(input: ParserInput)
-      extends Http4sHeaderParser[`Accept-Encoding`](input)
-      with ContentCodingParser {
-    def entry: Rule1[`Accept-Encoding`] =
-      rule {
-        oneOrMore(EncodingRangeDecl).separatedBy(ListSep) ~ EOL ~> { (xs: Seq[ContentCoding]) =>
-          `Accept-Encoding`(xs.head, xs.tail: _*)
-        }
-      }
+  private[http4s] val acceptEncodingParser: Parser1[`Accept-Encoding`] = {
+    import Rfc2616BasicRules._
+    import cats.parse.Parser._
+
+    (rep1Sep(ContentCoding.parser, 1, listSep) <* AdditionalRules.EOL).map(xs =>
```

Comment:
I don't _think_ we want the EOL on these.  If we were parsing multiple headers at once, we would have the `\r\n` be a separator between headers, and not consumed by this.

Reply:
Ok, EOL is removed